tests/admin/test_controller.py

```diff
@@ -4718,6 +4720,137 @@ def test_library_registrations_post_success(self):
+        collection.external_account_id = "collection url"
+
+        with self.app.test_request_context("/", method="POST"):
+            flask.request.form = MultiDict([
+                ("collection_id", collection.id),
+                ("library_short_name", "not-a-library"),
+            ])
+            response = self.manager.admin_settings_controller.collection_library_registrations()
+            eq_(COLLECTION_DOES_NOT_SUPPORT_REGISTRATION, response)
+
+        collection.protocol = SharedODLAPI.NAME
+
+        with self.app.test_request_context("/", method="POST"):
+            flask.request.form = MultiDict([
+                ("collection_id", collection.id),
+                ("library_short_name", "not-a-library"),
+            ])
+            response = self.manager.admin_settings_controller.collection_library_registrations()
+            eq_(NO_SUCH_LIBRARY, response)
+
+        library = self._default_library
+
+        with self.app.test_request_context("/", method="POST"):
+            flask.request.form = MultiDict([
+                ("collection_id", collection.id),
+                ("library_short_name", library.short_name),
+            ])
+            feed = '<feed></feed>'
+            self.responses.append(MockRequestsResponse(200, content=feed))
+
+            response = self.manager.admin_settings_controller.collection_library_registrations(do_get=self.do_request, do_post=self.do_request)
+            eq_(REMOTE_INTEGRATION_FAILED.uri, response.uri)
+            eq_("The service at collection url did not return OPDS.", response.detail)
+            eq_([collection.external_account_id], self.requests)
+            eq_("failure", ConfigurationSetting.for_library_and_externalintegration(
+                    self._db, "library-registration-status", library, collection.external_integration).value)
+
+        with self.app.test_request_context("/", method="POST"):
+            flask.request.form = MultiDict([
+                ("collection_id", collection.id),
+                ("library_short_name", library.short_name),
+            ])
+            feed = '<feed></feed>'
+            headers = { 'Content-Type': 'application/atom+xml;profile=opds-catalog;kind=navigation' }
+            self.responses.append(MockRequestsResponse(200, content=feed, headers=headers))
+
+            response = self.manager.admin_settings_controller.collection_library_registrations(do_get=self.do_request, do_post=self.do_request)
+            eq_(REMOTE_INTEGRATION_FAILED.uri, response.uri)
+            eq_("The service at collection url did not provide a register link.", response.detail)
+            eq_([collection.external_account_id], self.requests[1:])
+            eq_("failure", ConfigurationSetting.for_library_and_externalintegration(
+                    self._db, "library-registration-status", library, collection.external_integration).value)
+
+    def test_collection_library_registrations_post_success(self):
```

Comment:
You've already written these tests and there's no need to change them, but when refactoring admin controller code (the way you did with `_register_library`) I've found it useful to write a suite of tests for the helper method, and then much simpler tests for the methods that call the helper method, which just verify that it called that method with the right arguments.